### PR TITLE
Zqs 922 make qe qiskit tests run faster

### PR DESCRIPTION
**Pull request by AthenaCaesura**
A couple changes made here to speed up time.

- Pickled some data so we don't have to wait for IBM to respond
- Reduced batch size so that we aren't waiting for long processes to finish just for a test
- Reduced number of samples when possible
- deleted redundant tests
- lowered time between re-sumbitting jobs

Outcome: overall, tests run in about 1/3 of the time, but there is huge variability in how long we are waiting for IBM to run processes.

Perhaps the most useful info that I got out of this is that we should run tests either late in the night (around 2am EST) or in the afternoon (around 11am EST) becuase that appears to be when the servers have the least load. I can understand the 2am timeframe, but why no one is sending jobs at lunchtime is anyone's guess.

### CODE REVIEW

File: tests/qeqiskit/backend/backend_test.py [hunk 1]
```diff
@@ -132,7 +138,7 @@ def test_run_circuitset_and_measure(self, backend):
             # Then (since SPAM error could result in unexpected bitstrings, we make sure
             # the most common bitstring is the one we expect)
             counts = measurements.get_counts()
-            assert max(counts, key=counts.get) == "100"
+            assert max(counts, key=counts.get) == "1"
```
Comment (mstechly) [inline tests/qeqiskit/backend/backend_test.py:141]: Why should it be "1" now if we still are doing 100 samples?

File: tests/qeqiskit/backend/backend_test.py [hunk 2]
```diff
@@ -118,7 +124,7 @@ def test_aggregate_measurements(self, backend):
     def test_run_circuitset_and_measure(self, backend):
         # Given
         num_circuits = 10
-        circuit = self.x_cnot_circuit()
+        circuit = self.x_circuit()
```
Comment (mstechly) [inline tests/qeqiskit/backend/backend_test.py:127]: Why have you changed the circuit?

File: tests/qeqiskit/backend/backend_test.py [hunk 3]
```diff
@@ -177,36 +182,12 @@ def test_execute_with_retries_timeout(self, backend):
             for _ in range(num_jobs):
                 backend.execute_with_retries([circuit], n_samples)
 
-    def test_run_circuitset_and_measure_readout_correction_retries(
```
Comment (mstechly) [inline tests/qeqiskit/backend/backend_test.py:180]: Why deleting this test?

File: tests/qeqiskit/backend/backend_test.py (hunk 1 above)
Comment (AthenaCaesura) [reply tests/qeqiskit/backend/backend_test.py:141]: it cuts down on the computation time a bit. Beause you are using a simulator there is no risk of SPAM errors. 

Maybe I should remove that comment as it is confusing.

File: tests/qeqiskit/backend/backend_test.py (hunk 2 above)
Comment (AthenaCaesura) [reply tests/qeqiskit/backend/backend_test.py:127]: Consistency. Not really any other reason. Is there a reason to keep it the way it was?

File: tests/qeqiskit/backend/backend_test.py (hunk 3 above)
Comment (AthenaCaesura) [reply tests/qeqiskit/backend/backend_test.py:180]: It seemed like a dupilcate of test_readout_correction_works_run_circuitset_and_measure.

File: tests/qeqiskit/backend/backend_test.py (hunk 2 above)
Comment (mstechly) [reply tests/qeqiskit/backend/backend_test.py:127]: No, it was just unclear to me.

File: tests/qeqiskit/backend/backend_test.py (hunk 3 above)
Comment (mstechly) [reply tests/qeqiskit/backend/backend_test.py:180]: I don't think so – this one explicitly targets "a race condition". 
But to be honest it seems to me it's broken, as the comment says that "The test will still pass".
I think better option is to leave it as it, but add `xfail` or `skip` at the beginning.

**Review by mstechly — CHANGES_REQUESTED** [review]

**Review by mstechly — APPROVED** [review]

**Pull request merged**